internal/pkg/aws/cloudwatch/cloudwatch.go

```diff
@@ -125,6 +133,58 @@ func (cw *CloudWatch) AlarmStatuses(opts ...DescribeAlarmOpts) ([]AlarmStatus, e
 	return alarmStatuses, nil
 }
 
+// AlarmDescriptions returns the config of alarms filtered by name.
+func (cw *CloudWatch) AlarmDescriptions(alarmNames []string) ([]*AlarmDescription, error) {
+	var alarmDescriptions []*AlarmDescription
+	in := &cloudwatch.DescribeAlarmsInput{
+		AlarmNames: aws.StringSlice(alarmNames),
+	}
+	for {
+		alarmResp, err := cw.client.DescribeAlarms(in)
+		if err != nil {
+			return nil, fmt.Errorf("describe CloudWatch alarms: %w", err)
+		}
+		if alarmResp == nil {
+			break
+		}
+		alarmDescriptions = append(alarmDescriptions, cw.compositeAlarmsDescriptions(alarmResp.CompositeAlarms)...)
+		alarmDescriptions = append(alarmDescriptions, cw.metricAlarmsDescriptions(alarmResp.MetricAlarms)...)
+		if alarmResp.NextToken == nil {
+			break
+		}
+		in.NextToken = alarmResp.NextToken
+	}
+	return alarmDescriptions, nil
+}
+
+func (cw *CloudWatch) compositeAlarmsDescriptions(alarms []*cloudwatch.CompositeAlarm) []*AlarmDescription {
```

Comment:
how about
```suggestion
func (cloudwatch.CompositeAlarm) description() *AlarmDescription {
```
and move the for loop outside 🤔

Reply:
I think it's actually a little more streamlined and clear as is, and also avoids nested for-loops. 🤷🏼